File: Set1_1000/167_Two_SumII_Input_Array_Is_Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left_min = 0
        right_max = len(numbers) - 1
        while left_min < right_max:
            if numbers[left_min] + numbers[right_max] < target:
                left_min += 1
            elif numbers[left_min] + numbers[right_max] > target:
                right_max -= 1
            else:
                return [left_min+1, right_max+1]
        

File: Set2_1000/167_Two_Sum_II_Input_Array_Is_Sorted.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        if (nums == null || nums.length == 0) {
            return new int []{};
        }
        int length = nums.length;
        int left = 0;
        int right = length - 1;
        while (left <= right) {
            if (nums[left] + nums[right] == target) {
                return new int []{left + 1, right + 1};
            }
            if (nums[left] + nums[right] > target) {
                right--;
            }
            else {
                left++;
            }
        }
        return new int []{};
    }
}